cmd/ocitools/generate.go

```diff
@@ -0,0 +1,299 @@
+	cli.StringSliceFlag{Name: "bind", Usage: "bind mount directories src:dest:(rw,ro)"},
+	cli.StringSliceFlag{Name: "prestart", Usage: "path to prestart hooks"},
+	cli.StringSliceFlag{Name: "poststart", Usage: "path to poststart hooks"},
+	cli.StringSliceFlag{Name: "poststop", Usage: "path to poststop hooks"},
+	cli.StringFlag{Name: "root-propagation", Usage: "mount propagation for root"},
+	cli.StringFlag{Name: "os", Value: runtime.GOOS, Usage: "operating system the container is created for"},
+	cli.StringFlag{Name: "arch", Value: runtime.GOARCH, Usage: "architecture the container is created for"},
+	cli.StringFlag{Name: "cwd", Value: "/", Usage: "current working directory for the process"},
+	cli.StringSliceFlag{Name: "uidmappings", Usage: "add UIDMappings e.g HostID:ContainerID:Size"},
+	cli.StringSliceFlag{Name: "gidmappings", Usage: "add GIDMappings e.g HostID:ContainerID:Size"},
+	cli.StringSliceFlag{Name: "sysctl", Usage: "add sysctl settings e.g net.ipv4.forward=1"},
+	cli.StringFlag{Name: "apparmor", Usage: "specifies the the apparmor profile for the container"},
+	cli.StringFlag{Name: "seccomp-default", Usage: "specifies the the defaultaction of Seccomp syscall restrictions"},
+	cli.StringSliceFlag{Name: "seccomp-arch", Usage: "specifies Additional architectures permitted to be used for system calls"},
+	cli.StringSliceFlag{Name: "seccomp-syscalls", Usage: "specifies Additional architectures permitted to be used for system calls, e.g Name:Action:Arg1_index/Arg1_value/Arg1_valuetwo/Arg1_op, Arg2_index/Arg2_value/Arg2_valuetwo/Arg2_op "},
+	cli.StringSliceFlag{Name: "seccomp-allow", Usage: "specifies syscalls to be added to allowed"},
+	cli.StringSliceFlag{Name: "seccomp-errno", Usage: "specifies syscalls to be added to list that returns an error"},
+	cli.StringFlag{Name: "template", Usage: "base template to use for creating the configuration"},
+	cli.StringSliceFlag{Name: "label", Usage: "add annotations to the configuration e.g. key=value"},
+}
+
+var generateCommand = cli.Command{
+	Name:   "generate",
+	Usage:  "generate a OCI spec file",
+	Flags:  generateFlags,
+	Before: before,
+	Action: func(context *cli.Context) error {
+		// Start from the default template.
+		specgen := generate.New()
+
+		template := context.String("template")
+		if template != "" {
+			var err error
+			specgen, err = generate.NewFromFile(template)
+			if err != nil {
+				return err
+			}
+		}
+
+		err := modify(specgen, context)
+		if err != nil {
+			return err
+		}
+
+		if context.IsSet("output") {
+			output := context.String("output")
+			err = specgen.SaveToFile(output)
+		} else {
+			err = specgen.Save(os.Stdout)
+		}
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func modify(g generate.Generator, context *cli.Context) error {
+	spec := g.GetSpec()
+
+	if len(spec.Version) == 0 {
+		spec.Version = rspec.Version
+	}
+	spec.Root.Path = context.String("rootfs")
```

Comment:
We don't need get except for spec

Sent from my iPhone

> On Jul 15, 2016, at 7:12 AM, hmeng-19 <notifications@github.com> wrote:
> 
> In cmd/ocitools/generate.go:
> 
> > +			err = specgen.Save(os.Stdout)
> > +		}
> > +		if err != nil {
> > +			return err
> > +		}
> > +		return nil
> > +	},
> > +}
> > +
> > +func modify(g generate.Generator, context *cli.Context) error {
> > +	spec := g.GetSpec()
> > +
> > +	if len(spec.Version) == 0 {
> > +		spec.Version = rspec.Version
> > +	}
> > +	spec.Root.Path = context.String("rootfs")
> @mrunalp , Okay. In general, we can provide get/set (plus add if applicable) functions for each field inside g.spec. Right?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>